base/bootstrap.go

```diff
@@ -352,6 +355,9 @@ func (cc *CouchbaseCluster) WriteMetadataDocument(ctx context.Context, location,
 	if cc == nil {
 		return 0, errors.New("nil CouchbaseCluster")
 	}
+	if cas == 0 {
+		return 0, RedactErrorf("CAS for %q in bucket %q must be non-zero", UD(docID), MD(location))
```

Comment:
Including something in the error message about how the caller should be using InsertMetadataDocument if they are trying to insert might be helpful.